Eastern-Research-Group/Trade-off_Tool_for_Sampling
# Feature/63 update dependencies

## Related Issues:
* [TOTS-63](https://ergcloud.atlassian.net/browse/TOTS-63)

## Main Changes:
* Updated 3rd party dependencies across the app, including the ArcGIS JS API.
  * I already ran all of the cypress tests and they are passing.
  * The tooltip had a major update, which required code changes.
  * The ArcGIS JS API had some breaking changes that required code changes. This mostly affected the popups and sketching. 
  * With the cypress upgrade I ran into some issues with screenshots not working. I found this was do to a cypress bug in version `2.15.0`, so I only upgraded it to `2.14.0`. More info here https://github.com/cypress-io/cypress/issues/27097.
* Added layer name and layer id columns to the excel download.
* Updated cypress tests to accomodate recent changes.

## Steps To Test:
1. From the `app` folder run `npm run setup`
2. Start the app
3. Try to test most features throughout the app. Be sure to do some manual tests that involve plotting samples in both 2d and 3d.
4. Test downloading from the calculate resources tab and verify the `Layer Name` and `Layer ID` columns are available on the `Sample Details` tab of the download.
5. Run cypress tests and verify they all pass
